### PR TITLE
[Merged by Bors] - Fix building project when `go-svm` is imported

### PR DESCRIPTION
**Pull request by nkryuchkov**
## Motivation

WIP on https://github.com/spacemeshos/pm/issues/89:
> Add `go-svm` to `go-spacemesh` as a dependency (mostly about getting the build/linkage to work).

## Changes

- Import github.com/spacemeshos/go-svm/svm
- Fix Makefiles for the import

## Test Plan

UT

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by nkryuchkov** [comment]
TODO: create a test release from this branch and check if it works

**Comment by nkryuchkov** [comment]
`dockerpush` failed: https://github.com/spacemeshos/go-spacemesh/runs/4791511767?check_suite_focus=true
```
# github.com/spacemeshos/go-svm/svm
/go/src/github.com/spacemeshos/go-spacemesh/build//libsvm.a(sqlite3.o):(.data.rel.aSyscall+0xb0): undefined reference to `fcntl64'
collect2: error: ld returned 1 exit status
make: *** [go-spacemesh] Error 2
Makefile:105: recipe for target 'go-spacemesh' failed
The command '/bin/sh -c make build' returned a non-zero code: 2
make: *** [Makefile:209: dockerbuild-go] Error 2
Error: Process completed with exit code 2.
```

**Comment by nkryuchkov** [comment]
`go-spacemesh` in the release for macOS seems working (https://github.com/spacemeshos/go-spacemesh/releases/tag/untagged-c3a6339bc6579db217e5)

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
bors try

**Comment by nkryuchkov** [comment]
> `dockerpush` failed: https://github.com/spacemeshos/go-spacemesh/runs/4791511767?check_suite_focus=true
> 
> ```
> # github.com/spacemeshos/go-svm/svm
> /go/src/github.com/spacemeshos/go-spacemesh/build//libsvm.a(sqlite3.o):(.data.rel.aSyscall+0xb0): undefined reference to `fcntl64'
> collect2: error: ld returned 1 exit status
> make: *** [go-spacemesh] Error 2
> Makefile:105: recipe for target 'go-spacemesh' failed
> The command '/bin/sh -c make build' returned a non-zero code: 2
> make: *** [Makefile:209: dockerbuild-go] Error 2
> Error: Process completed with exit code 2.
> ```

fixed

**Comment by nkryuchkov** [comment]
New release https://github.com/spacemeshos/go-spacemesh/releases/tag/untagged-fea03f4b289681e3b984 seems working on macOS too

File: Dockerfile [hunk 1]
```diff
@@ -93,6 +96,7 @@ COPY --from=server_builder /go/src/github.com/spacemeshos/go-spacemesh/build/go-
 COPY --from=server_builder /go/src/github.com/spacemeshos/go-spacemesh/build/go-p2p /bin/
 COPY --from=server_builder /go/src/github.com/spacemeshos/go-spacemesh/build/go-harness /bin/
 COPY --from=server_builder /go/src/github.com/spacemeshos/go-spacemesh/build/libgpu-setup.so /bin/
+COPY --from=server_builder /go/src/github.com/spacemeshos/go-spacemesh/build/libsvm.so /bin/
```
Comment (dshulyak) [inline Dockerfile]: is it large? not sure if we should add it to the image given that it won't be used in near future (6m at least?)

Comment (nkryuchkov) [reply Dockerfile]: 13,7 MB

Comment (nkryuchkov) [reply Dockerfile]: I can comment it out until we need it
cc @moshababo

Comment (moshababo) [reply Dockerfile]: Yes, plz comment it out for now.

**Review by neysofu — APPROVED** [review]
:100:

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
> Build failed:
> 
> * [ci-stage2](https://github.com/spacemeshos/go-spacemesh/runs/4883451276?check_suite_focus=true)

looks like an infra failure

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**